src/ucp/core/ucp_ep.c

```diff
@@ -205,13 +243,14 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker,
 
     ep = ucp_worker_ep_find(worker, dest_uuid);
     if (ep != NULL) {
-        /* TODO: handle a case where:
-         * 1) the existing endpoint is incomplete
-         * 2) configuration (key) of existing endpoint does not match requested
-         *    parameters, for example, error handling level
-         */
-        *ep_p = ep;
-        status = UCS_OK;
+        status = ucp_ep_adjust_params(ep, params);
+        if (status == UCS_OK) {
+            *ep_p = ep;
+        } else {
+            ucs_error("ep configuration mismatch with error: %s",
```

Comment:
i think there is no need to print an error here, since ucp_ep_adjust_params() already prints error for all cases..
can just do if (status != UCS_OK) { return status; }